include/seastar/net/tls.hh

```diff
@@ -330,9 +330,14 @@ namespace tls {
     ::seastar::socket socket(shared_ptr<certificate_credentials>, sstring name = {});
     /// @}
 
-    /** Wraps an existing connection in SSL/TLS. */
+    /** 
+     * Wraps an existing connection in SSL/TLS.
+     *
+     * \param wait_for_eof Optionally indicate to the client if it should wait for an EOF from the server
+     * after sending the BYE message. By default, the wait is hardcoded to 10 seconds.
+     */
     /// @{
-    future<connected_socket> wrap_client(shared_ptr<certificate_credentials>, connected_socket&&, sstring name = {});
+    future<connected_socket> wrap_client(shared_ptr<certificate_credentials>, connected_socket&&, sstring name = {}, bool wait_for_eof = true);
```

Comment:
While perhaps overengineering, I think this should be something like:
```
struct tls_options {
   bool wait_for_eof_on_shutdown = true;
   // add as needed.
};

 future<connected_socket> wrap_client(shared_ptr<certificate_credentials>, connected_socket&&, sstring name = {}, std::optional<tls_options> = {});

```
and from there to initing a tls_options to this or defaults in session.

It would make it easier to extend with any further required options ahead without breaking API.

Reply:
Good shout. I kept the change super minimal to begin with as I was unsure what you'd think about it. Updating the PR.

Reply:
Re-tested and force pushed the update.